app/models/dossier.rb

```diff
@@ -522,16 +525,42 @@ def messagerie_available?
     !brouillon? && !user_deleted? && !archived
   end
 
-  def en_construction_close_to_expiration?
-    self.class.en_construction_close_to_expiration.exists?(id: self)
+  def close_to_expiration_at
+    if brouillon?
+      created_at
+    elsif en_construction?
+      en_construction_at
+    elsif en_instruction?
+      en_instruction_at
```

Comment:
note mfo : demander prk d'un coté on considérère qu'il n'est pas expirable en instruction, de l'autre on le dossier retourner une date d'expiraiton.

Reply:
Excellente question :) Je suis :+1: pour ne pas retourner de date d'expiration en_instruction pour l'instant. La raison pour laquelle on n'expire pas les dossiers en_instruction c'est qu'on ne sait pas quoi faire légalement.

Reply:
impec! sur ces cas "à la marge" ; je suis tenté de ` else raise 'not supported, we do not know what to do with ...'` ; ca evite d'appeler le code dans un usage non souhaité, des cas a la marge etrange etc...